example/src/main/java/io/netty/example/http2/helloworld/frame/server/HelloWorldHttp2Handler.java

```diff
@@ -70,18 +71,17 @@ public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
      * If receive a frame with end-of-stream set, send a pre-canned response.
      */
     private static void onDataRead(ChannelHandlerContext ctx, Http2DataFrame data) throws Exception {
-        int consumed = data.padding() + data.content().readableBytes();
-        int streamId = data.streamId();
+        Http2FrameStream stream = data.stream();
 
         if (data.isEndStream()) {
-            sendResponse(ctx, streamId, data.content());
+            sendResponse(ctx, stream, data.content());
         } else {
             // We do not send back the response to the remote-peer, so we need to release it.
             data.release();
         }
 
         // Update the flowcontroller
-        ctx.write(new DefaultHttp2WindowUpdateFrame(consumed).streamId(streamId));
+        ctx.write(new DefaultHttp2WindowUpdateFrame(data.initialFlowControlledBytes()).stream(stream));
```

Comment:
I wonder if we should have a constructor which accepts the stream ID for some frames ... seems like it maybe less noisy in user code ... and *maybe* folks would be less likely to forget?